lib/plugins/block_actions.js

```diff
@@ -2,19 +2,72 @@ const Vec3 = require('vec3').Vec3
 
 module.exports = inject
 
+const CARDINALS = {
+  north: new Vec3(0, 0, -1),
+  south: new Vec3(0, 0, 1),
+  west: new Vec3(-1, 0, 0),
+  east: new Vec3(1, 0, 0)
+}
+
 function inject (bot, { version }) {
   const { instruments, blocks } = require('minecraft-data')(version)
+
+  function parseChestMetadata (chestBlock) {
+    const chestTypes = ['single', 'right', 'left']
+
+    return bot.supportFeature('doesntHaveChestType')
+      ? { facing: Object.keys(CARDINALS)[chestBlock.metadata - 2] }
+      : {
+        waterlogged: !(chestBlock.metadata & 1),
+        type: chestTypes[(chestBlock.metadata >> 1) % 3],
+        facing: Object.keys(CARDINALS)[Math.floor(chestBlock.metadata / 6)]
+      }
+  }
+
+  function getChestType (chestBlock) { // Returns 'single', 'right' or 'left'
+    if (bot.supportFeature('doesntHaveChestType')) {
+      const facingMap = {
+        north: { west: 'right', east: 'left' },
+        south: { west: 'left', east: 'right' },
+        west: { north: 'left', south: 'right' },
+        east: { north: 'right', south: 'left' }
+      }
+
+      const facing = parseChestMetadata(chestBlock).facing
+
+      // We have to check if the adjacent blocks in the perpendicular cardinals are the same type
+      const perpendicularCardinals = Object.keys(facingMap[facing])
+      for (const cardinal of perpendicularCardinals) {
+        const cardinalOffset = CARDINALS[cardinal]
```

Comment:
This can be compressed to 
```js
for (const [cardinal, cardinalOffset] of Object.entries(CARDINALS)) {
```

Reply:
I changed it a bit so it only checks left and right and not the front and back too

Reply:
Hmmm.. You're right